controller/api/destination/server.go

```diff
@@ -7,23 +7,28 @@ import (
 	"strings"
 
 	pb "github.com/linkerd/linkerd2-proxy-api/go/destination"
-	"github.com/linkerd/linkerd2/controller/api/util"
+	"github.com/linkerd/linkerd2/controller/api/destination/watcher"
 	discoveryPb "github.com/linkerd/linkerd2/controller/gen/controller/discovery"
 	"github.com/linkerd/linkerd2/controller/k8s"
-	"github.com/linkerd/linkerd2/pkg/addr"
 	"github.com/linkerd/linkerd2/pkg/prometheus"
-	log "github.com/sirupsen/logrus"
+	logging "github.com/sirupsen/logrus"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/peer"
 )
 
-type server struct {
-	k8sAPI          *k8s.API
-	resolver        streamingDestinationResolver
-	enableH2Upgrade bool
-	controllerNS,
-	identityTrustDomain string
-	log *log.Entry
-}
+type (
+	server struct {
+		endpoints *watcher.EndpointsWatcher
+		profiles  *watcher.ProfileWatcher
+
+		enableH2Upgrade     bool
+		controllerNS        string
+		identityTrustDomain string
```

Comment:
All of these things are endpoints-specific, right? Can this configuration be pushed into the EndpointsWatcher?

Reply:
no, they're configured on the destination server by the flags:
* `enable-h2-upgrade`
* `controller-namespace`
* and the global identity context